marc_to_solr/lib/access_facet_builder.rb

```diff
@@ -0,0 +1,64 @@
+# Class for building access_facet values
+class AccessFacetBuilder
+  # Build access facet
+  # @param record [MARC::Record]
+  # @param context [Traject::Indexer::Context]
+  # @return [Array<String>] access values
+  def self.build(record:, context:)
+    new(record: record, context: context).build
+  end
+
+  attr_reader :record, :context
+
+  # Constructor
+  # @param record [MARC::Record]
+  # @param context [Traject::Indexer::Context]
+  def initialize(record:, context:)
+    @record = record
+    @context = context
+  end
+
+  # @return [Array<String>] access values
+  def build
+    [
+      electronic_portfolio,
+      in_library,
+      hathi,
+      hathi_temp,
+      marc_indicator
+    ].uniq.compact
+  end
+
+  private
+
+    def electronic_portfolio
+      return 'Online' if context.output_hash['electronic_portfolio_s'].present?
+    end
+
+    def in_library
+      return 'In the Library' if context.output_hash['location_code_s'].present?
+    end
+
+    def hathi
+      return 'Online' if hathi_locations.include?('hathi')
+    end
+
+    def hathi_locations
+      return [] unless context.output_hash['hathi_identifier_s'].present?
+      hathi_line = find_hathi_by_oclc(context.output_hash['oclc_s'].first)
+      parse_locations_from_hathi_line(hathi_line)
+    end
+
+    def hathi_temp
+      return 'Temporary Digital Access' if hathi_locations.include?('hathi_temp')
+    end
```

Comment:
We don't have the 'Temporary Digital Access' facet. It should not appear in the catalog. catalog staging has it because of an old index. @kevinreiss @mzelesky

Reply:
I'm not sure if it matters and if there is a reason to keep it. I looked in bibdata-worker2 under `shared/hathi/input and output` and we dont have the hathi file with the 'deny', 'allow' access anymore. Now I start to remember we used the 'Temporary Digital access' facet to index and separate the deny 'hathi' records but we never used it because the hathi access changed. So I think my question is more for @mzelesky and @kevinreiss can we remove this 'Temporary Digital Access' facet from the code?

Reply:
Yes, we can remove this. We never finished implementing this feature and even if we are asked to extend beyond the Voyager migration I do not think we would ever try to use this again.